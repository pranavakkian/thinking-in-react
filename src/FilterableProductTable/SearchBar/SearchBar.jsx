import React, { memo } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  handleFilterTextChange,
  handleIsStockOnlyChange,
} from '../Redux/SearchBarReducer/Actions/Actions';

const SearchBar = () => {
  const dispatch = useDispatch();

  return (
    <>
      <input
        type="text"
        value={useSelector((state) => state.SearchBarReducer.filterText)}
        onChange={(event) => dispatch(handleFilterTextChange(event.target.value))}
        name="filterText"
        id="filterText"
        placeholder="Search product..."
      />
      <br />
      <input
        checked={useSelector((state) => state.SearchBarReducer.isStockOnly)}
        onChange={(event) => dispatch(handleIsStockOnlyChange(event.target.checked))}
        type="checkbox"
        name="isStockOnly"
        id="isStockOnly"
      />
      <span id="isStockOnlyLabel">Only show product in stock</span>
    </>
  );
};

export default memo(SearchBar);
