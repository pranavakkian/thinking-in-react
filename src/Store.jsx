import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import RootReducer from './RootReducer';

const composeEnhancers = composeWithDevTools({ trace: true });
const Store = createStore(RootReducer, composeEnhancers(applyMiddleware()));

export default Store;
