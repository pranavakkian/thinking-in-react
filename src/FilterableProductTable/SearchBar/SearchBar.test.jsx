/* eslint-disable no-undef */
import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import Store from '../../Store';
import SearchBar from './SearchBar';

const wrapper = mount(
  <Provider store={Store}>
    <SearchBar />
  </Provider>,
);

describe('SearchBar', () => {
  it('SearchBar rendered successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Number of input elements', () => {
    const inputElement = wrapper.find('input');
    expect(inputElement).toHaveLength(2);
  });

  it('Number of span elements', () => {
    const spanElements = wrapper.find('span');
    expect(spanElements).toHaveLength(1);
  });

  it('Checkbox working fine', () => {
    let checkbox = wrapper.find('#isStockOnly');
    expect(checkbox.props().checked).toEqual(false);
    checkbox.simulate('change', { target: { checked: true } });
    checkbox = wrapper.find('#isStockOnly');
    expect(checkbox.props().checked).toEqual(true);
  });

  it('Search input field working fine', () => {
    let filterText = wrapper.find('#filterText');
    expect(filterText.props().value).toEqual('');
    expect(filterText.props().placeholder).toEqual('Search product...');
    filterText.simulate('change', { target: { value: 'hello' } });
    filterText = wrapper.find('#filterText');
    expect(filterText.props().value).toEqual('hello');
  });

  it('Checkbox label', () => {
    const checkboxLabel = wrapper.find('#isStockOnlyLabel');
    expect(checkboxLabel.text()).toEqual('Only show product in stock');
  });

  it('Search bar un-mounted successfully', () => {
    wrapper.unmount();
  });
});
