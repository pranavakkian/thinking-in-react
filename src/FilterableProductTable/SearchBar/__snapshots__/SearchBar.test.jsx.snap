// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar SearchBar rendered successfully 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Memo(SearchBar)>
    <input
      id="filterText"
      name="filterText"
      onChange={[Function]}
      placeholder="Search product..."
      type="text"
      value=""
    />
    <br />
    <input
      checked={false}
      id="isStockOnly"
      name="isStockOnly"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      id="isStockOnlyLabel"
    >
      Only show product in stock
    </span>
  </Memo(SearchBar)>
</Provider>
`;
